.github/workflows/ts-packages.yml

```diff
@@ -60,6 +60,53 @@ jobs:
           ENABLE_GAS_REPORT: 1
       - name: Print gas report
         run: cat packages/contracts/gas-report.txt
+
+  test-coverage:
+    name: Generate test coverage
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Fetch history
+        run: git fetch
+
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install Dependencies
+        # only install dependencies if there was a change in the deps
+        # if: steps.yarn-cache.outputs.cache-hit != 'true'
+        run: yarn install
+
+      - name: Build
+        run: yarn build
+
+      - name: Test Coverage
+        run: yarn test:coverage
+        # Tests are quite brittle when instrumented by Solidity Coverage.
+        # If tests pass in the Test job above, tests failing during coverage shouldn't break CI.
+        continue-on-error: true
+      - uses: codecov/codecov-action@v1
+        with:
+          files: ./packages/contracts/coverage.json
+          fail_ci_if_error: false
+          verbose: true
```

Comment:
We should upload the coverage results into the workflow artefacts

Reply:
Sounds good to me, can you point me to docs about how to achieve that?

Reply:
It's [here](https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts) although I want to see `codechecks` output as that might already be covering it. I tried viewing the example report in your repo but got an error `Sorry, we couldn't find that page`

Reply:
> I tried viewing the example report in your repo but got an error Sorry, we couldn't find that page

Assuming you mean the one I mentioned [here](https://github.com/ethereum-optimism/optimism/pull/732#issuecomment-831258181), it's a public repo, so I'm not sure why it doesn't work, but it's exactly the same as the bot comment [here](https://github.com/ethereum-optimism/optimism/pull/732#issuecomment-831259394).

Reply:
I see it now but only lines coverage. We really need branch coverage through, does it support that?